content/en/blog/2022-07-05-unikraft-gsoc-intel-sgx.md

```diff
@@ -0,0 +1,32 @@
+
+Consider this from a practical application scenario, if the tenant does not trust the cloud service provider, then with the presence of TEE, the tenant can isolate the sensitive information and code they have from other data via TEE (Intel SGX), thus making it impossible for the cloud service provider to access the data that needs to be protected. Unikraft is committed to providing efficient and secure cloud services to its users, thus, TEE is a must for Unikraft.
+
+## How to?
+Thanks to the open-source community, we have lots of reference to learn from, such as [linux-sgx-driver](https://github.com/intel/linux-sgx-driver) and [freebsd-sgx](https://github.com/bukinr/freebsd-sgx). Enabling Intel SGX in Unikraft is a huge task, as we need to port both the driver and the SDK to the Unikraft platform. Although drivers and SDKs on different OSs are open-sourced, we still need to significantly modify them, which removes the OS-specific things and converts them to Unikraft-specific features and dependencies.
+
+## Progress
+Currently, we are working on porting the driver into Unikraft. As of now, the following functions are finished:
+
+- The probing of SGX. That is to say, Unikraft can be aware of SGX if enabled in `make menuconfig`. And as inspired by linux-sgx-driver, the uksgx library registers a device /dev/sgx to the devfs, and some stub ioctl function is added to the library, in order to provide a unified and easy interaction channel between uksgx and the applications. But please notice that ioctl is not the Unikraft-ish way to interact with devices/internal-libs. Now, that's only for the test and compatibility with the existing codes and SDKs and will be refactored to more straightforward implementation.
```

Comment:
```suggestion
- The probing of SGX. That is to say, Unikraft can be aware of SGX if enabled in `make menuconfig`. And as inspired by `linux-sgx-driver`, the `uksgx` library registers a device `/dev/sgx` to `devfs`, and some stub `ioctl()` function is added to the library, in order to provide a unified and easy interaction channel between `uksgx` and applications. But please notice that `ioctl` is not the Unikraft-ish way to interact with devices / internal-libs. At this point, this is only for testing and for compatibility with the existing codes and SDKs; it will be refactored to more straightforward implementation.
```